mapper/orm/src/main/java/org/hibernate/search/mapper/orm/search/loading/impl/HibernateOrmByIdEntityLoader.java

```diff
@@ -133,12 +134,34 @@ else if ( hasExpectedType( reference, loadedEntity ) ) {
 		return (List<E>) loadedEntities;
 	}
 
-	private MultiIdentifierLoadAccess<?> getMultiAccess() {
+	private List<?> multiLoad(List<Serializable> ids, Integer timeout) {
 		MultiIdentifierLoadAccess<?> multiAccess = session.byMultipleIds( targetEntityType.getEntityName() );
-
 		multiAccess.withBatchSize( loadingOptions.getFetchSize() );
+		if ( timeout == null ) {
+			return multiAccess.multiLoad( ids );
+		}
+
+		// Multi access API does not allow to set a timeout invoking a multiLoad.
+		// Wrapping the loading within a transaction so that we can set the timeout there.
+		List<?> result;
+		Transaction tx = session.beginTransaction();
```

Comment:
Woah there, hold on. You're starting a transaction like it was nothing. What if a transaction is already open? What if the user doesn't want to use transactions? What if the user has JTA enabled, and transactions cannot be opened this way?

Opening a transaction yourself is definitely not an option. Even if it was, opening one transaction per type to load is really not a good solution.

Also, setting the timeout like this probably does nothing. As stated in the javadoc, calling `setTimeout()` will "Set the transaction timeout for any transaction started by a subsequent call to 'begin' on this instance.". Since `begin()` was already called when you call `setTimeout()`, it won't work.

The thing is, when Hibernate Search comes into play, there may already be a transaction, so I think it's too late to set the timeout. And if there is no transaction, it's likely the user wants it to stay that way. As you are well aware, we do have support for using Hibernate Search without transactions (though it's obviously not recommended).

Really, until they implement timeouts/hints in multiAccess, if we want timeouts in here, I think we're out of options. We may have to use criteria queries instead. Let's talk?